electroncash/tests/test_paymentrequests.py

```diff
@@ -144,3 +150,101 @@ def start_serve(self):
+    response = Response()
+    response.status_code = 200
+    response._content = str.encode(json.dumps(payment_req_data))
+    response.headers['digest'] = 'SHA-256=%s' % sha256(response._content).hex()
+    _, privkey, _ = deserialize_privkey('KzMFjMC2MPadjvX5Cd7b8AKKjjpBSoRKUTpoAtN6B3J9ezWYyXS6')
+    key = regenerate_key(privkey)
+    response.headers['signature'] = key.sign(sha256(response._content)).hex()
+    response.headers['x-identity'] = '17azqT8T16coRmWKYFj3UjzJuxiYrYFRBR'
+    response.url = 'https://bitpay.com/i/test123'
+    return response
+
+class Test_PaymentRequests_BitPay(unittest.TestCase):
+    def mocked_bitpay_requests_get_correct_sig(*args, **kwargs):
+        assert 'application/payment-request' == kwargs['headers']['accept']
+        response = _signed_response()
+        return response
+
+    # Verify that a payment request verification is successfull
+    @mock.patch('requests.get', side_effect=mocked_bitpay_requests_get_correct_sig)
+    def test_bitpay_verify(self, mock_get):
+        pr = get_payment_request('https://bitpay.com/invoice')
+        self.assertTrue(pr.error is None)
+        self.assertEquals(pr.get_memo(), 'dummy_memo')
+        self.assertEquals(pr.get_payment_url(), 'https://bitpay.com/i/test123')
+        self.assertTrue(pr.verify(None))
+        self.assertEquals(pr.get_requestor(), 'BitPay, Inc.')
+
+    def mocked_bitpay_requests_get_incorrect_digest(*args, **kwargs):
+        response = _signed_response()
+        # Trash digest by changing last char
+        response.headers['digest'] = response.headers['digest'][:-1] + '0'
+        return response
+
+    # Verify that a payment request verification fails if the digest is incorrect
+    @mock.patch('requests.get', side_effect=mocked_bitpay_requests_get_incorrect_digest)
+    def test_bitpay_verify_incorrect_digest(self, mock_get):
+        pr = get_payment_request('https://bitpay.com/invoice')
+        self.assertTrue(pr.error is None)
+        self.assertEquals(pr.get_memo(), 'dummy_memo')
+        self.assertEquals(pr.get_payment_url(), 'https://bitpay.com/i/test123')
+        self.assertFalse(pr.verify(None))
+
+    def mocked_bitpay_requests_get_incorrect_signature(*args, **kwargs):
+        response = _signed_response()
+        # Trash signature by changing last char
+        response.headers['signature'] = response.headers['signature'][:-1] + '0'
+        return response
+
+    # Verify that a payment request verification fails if the signature is incorrect
+    @mock.patch('requests.get', side_effect=mocked_bitpay_requests_get_incorrect_signature)
+    def test_bitpay_verify_incorrect_signature(self, mock_get):
+        pr = get_payment_request('https://bitpay.com/invoice')
+        self.assertTrue(pr.error is None)
+        self.assertEquals(pr.get_memo(), 'dummy_memo')
+        self.assertEquals(pr.get_payment_url(), 'https://bitpay.com/i/test123')
+        self.assertFalse(pr.verify(None))
```

Comment:
This is broken now.  Fails here on tox. Can u fix? Run tox to see what's happening.